src/Rules/Properties/AccessStaticPropertiesRule.php

```diff
@@ -49,11 +49,19 @@ public function getNodeType(): string
 
 	public function processNode(Node $node, Scope $scope): array
 	{
-		if (!$node->name instanceof Node\VarLikeIdentifier) {
+		if (
+			!$node->name instanceof Node\VarLikeIdentifier
+			&&
+			!$node->name instanceof Node\Expr\Variable
```

Comment:
This doesn't make much sense to me. The StaticPropertyFetch::$name can be `VarLikeIdentifier|Expr`. Why make special case for `Variable`?

Reply:
Variable variable like ```static::${$foo}``` are from type ```Node\Expr\Variable```